pkg/cmd/proxy/probe/exec.go

```diff
@@ -0,0 +1,324 @@
+	}
+
+	w := newWriter(streams)
+	for _, cluster := range managedClusterList.Items {
+		if err := o.visit(&w, hubRestConfig, addonClient, tunnel.DialContext, cluster.Name); err != nil {
+			klog.Errorf("An error occurred when requesting: %v", err)
+		}
+	}
+
+	w.flush()
+	return nil
+}
+
+const (
+	inClusterSecretProxyCA = "proxy-server-ca"
+	inClusterSecretClient  = "proxy-client"
+)
+
+func (o *Options) getKonnectivityTLSConfig(proxyConfig *proxyv1alpha1.ManagedProxyConfiguration) (*tls.Config, error) {
+	if o.isProxyClientCertProvided {
+		// building tls config from local paths
+		tlsCfg, err := proxyutil.GetClientTLSConfig(
+			o.proxyClientCACertPath,
+			o.proxyClientCertPath,
+			o.proxyClientKeyPath,
+			o.proxyServerHost, nil)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed building TLS config for konnectivity client")
+		}
+		return tlsCfg, nil
+	}
+	// building tls config from secret data
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed building cilent config")
+	}
+	nativeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed building cilent")
+	}
+	caSecret, err := nativeClient.CoreV1().Secrets(proxyConfig.Spec.ProxyServer.Namespace).
+		Get(context.TODO(), inClusterSecretProxyCA, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting CA secret")
+	}
+	caData := caSecret.Data["ca.crt"]
+	certSecret, err := nativeClient.CoreV1().Secrets(proxyConfig.Spec.ProxyServer.Namespace).
+		Get(context.TODO(), inClusterSecretClient, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting cert & key secret")
+	}
+	certData := certSecret.Data["tls.crt"]
+	keyData := certSecret.Data["tls.key"]
+
+	// building tls config from pem-encoded data
+	tlsCfg, err := buildTLSConfig(caData, certData, keyData, "127.0.0.1", nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed building TLS config from secret")
+	}
+	return tlsCfg, nil
+
+}
+
+func (o *Options) visit(
+	w *writer,
+	hubRestConfig *rest.Config,
+	addonClient addonv1alpha1client.Interface,
+	dialFunc k8snet.DialFunc,
+	clusterName string) error {
+
+	addon, err := addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterName).
+		Get(context.TODO(), common.AddonName, metav1.GetOptions{})
+	installed := "True"
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed getting managed cluster addon for cluster %v", clusterName)
+		}
+		installed = "False"
+	}
+	addonAvailable := "False"
+	for _, cond := range addon.Status.Conditions {
+		if cond.Type == addonv1alpha1.ManagedClusterAddOnConditionAvailable {
+			addonAvailable = string(cond.Status)
+		}
+	}
+
+	copiedCfg := rest.CopyConfig(hubRestConfig)
+	copiedCfg.Dial = dialFunc
+	copiedCfg.TLSClientConfig = rest.TLSClientConfig{}
+	copiedCfg.TLSClientConfig.Insecure = true
```

Comment:
maybe add a comment or TODO here

Reply:
adding a TODO comment and elaboration about why insecure TLS is needed here